tests/failpoints/cases/test_conf_change.rs

```diff
@@ -316,3 +316,101 @@ fn test_handle_conf_change_when_apply_fsm_resume_pending_state() {
         cluster.must_put(format!("kk{}", i).as_bytes(), b"v1");
     }
 }
+
+// This test simulates a scenario where a configuration change has been applied
+// on the target peer, allowing a leader transfer to that peer even if the
+// change hasn't been applied on the current leader.
+//
+// The setup involves a 4-node cluster where peer-1 starts as the leader. A
+// configuration change is initiated to remove peer-2. This change commits
+// successfully but only applies on peer-2 and peer-4.
+//
+// The expected result for leader transfer is:
+//   - It will fail to peer-2 because it has been removed.
+//   - It will fail to peer-3 because it has unapplied configuration change.
+//   - It will succeed to peer-4 because it has already applied the
+//     configuration change.
+#[test]
+fn test_applied_conf_change_on_target_peer_allows_transfer_leader() {
+    let mut cluster = new_server_cluster(0, 4);
+    let pd_client = cluster.pd_client.clone();
+    pd_client.disable_default_operator();
+    let region_id = cluster.run_conf_change();
+    pd_client.must_add_peer(region_id, new_peer(2, 2));
+    pd_client.must_add_peer(region_id, new_peer(3, 3));
+    pd_client.must_add_peer(region_id, new_peer(4, 4));
+
+    cluster.must_put(b"k1", b"v1");
+
+    fail::cfg("apply_on_conf_change_1_1", "pause").unwrap();
+    fail::cfg("apply_on_conf_change_3_1", "pause").unwrap();
+
+    pd_client.remove_peer(region_id, new_peer(2, 2));
+    // Wait for remove_peer.
+    sleep_ms(300);
+    // Peer 2 is still exists since the leader hasn't applied the ConfChange
+    // yet.
+    pd_client.must_have_peer(region_id, new_peer(2, 2));
+
+    // Use async_put for insertion here to avoid timeout errors, as synchronize put
+    // would hang due to the leader's apply process being paused.
+    let _ = cluster.async_put(b"k2", b"v2");
```

Comment:
```suggestion
    let _ = cluster.async_put(b"k2", b"v2").unwrap();
```
Pls use `unwrap` to ensure that the write is valid.